include/dynamatic/Dialect/Handshake/HandshakeOps.td

```diff
@@ -1008,6 +1009,7 @@ def SpeculatorOp : Handshake_Op<"speculator", [
 }
 
 def SpecSaveOp : Handshake_Op<"spec_save", [
+  SpeculationOpInterface,
```

Comment:
Since you have the interface (good idea btw!) you can now remove the
```tablegen
TypesMatchWith<"data operand type matches data result type",
                 "dataIn", "dataOut", "$_self">] >
```
trait from all concrete speculation operations.

Reply:
After taking another look, the slightly different trait definition for `SpeculatingBranchOp` makes my suggestion inconvenient, so let's forget about this.